File: LeetCodePython/py26.py
# 删除排序数组中的重复项
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index = 0
        result = 0
        while index<len(nums):
            if index == 0  or nums[index] != nums[index-1]:
                nums[result] = nums[index]
                result += 1
            index += 1
        return result
        

File: LeetCodeCPlusPlus/py26.cpp
#include<vector>
using namespace std;

// 删除排序数组中的重复项。与前面的不同，为空则插入应该插入的位置，插入位置+1

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int index=0,result = 0;
        while (index<nums.size()) {
            if (index==0||nums[index]!=nums[index-1]) {
                nums[result++] = nums[index];
            }
            index++;
        }
        return result;
    }
};